test/server/cards/09.1-FtE/HirumaSignaller.spec.js

```diff
@@ -0,0 +1,134 @@
+                this.player2.clickCard(this.hirumaSignaller);
+                expect(this.player2).toHavePrompt('Hiruma Signaller');
+                expect(this.player2).toBeAbleToSelect(this.hidaYakamo);
+                this.player2.clickCard(this.hidaYakamo);
+                expect(this.hidaYakamo.bowed).toBeFalsy();
+                expect(this.hidaYakamo.inConflict).toBeTruthy();
+                expect(this.hirumaSignaller.location).toBe('dynasty discard pile');
+            });
+
+            it('should mova a ready character to the conflict', function() {
+                this.initiateConflict({
+                    attackers: [this.mirumotoRaitsugu],
+                    defenders: [this.hirumaSignaller]
+                });
+                this.player2.clickCard(this.hirumaSignaller);
+                expect(this.player2).toHavePrompt('Hiruma Signaller');
+                expect(this.player2).toBeAbleToSelect(this.hidaYakamo);
+                this.player2.clickCard(this.hidaYakamo);
+                expect(this.hidaYakamo.bowed).toBeFalsy();
+                expect(this.hidaYakamo.inConflict).toBeTruthy();
+                expect(this.hirumaSignaller.location).toBe('dynasty discard pile');
+            });
+
+            it('should not ready a character at home and move to the conflict if the sacrifice is prevented', function() {
+                this.hidaYakamo.bow();
+                this.initiateConflict({
+                    attackers: [this.mirumotoRaitsugu],
+                    defenders: [this.hirumaSignaller]
+                });
+                this.player2.playAttachment(this.reprieve, this.hirumaSignaller);
+                this.player1.pass();
+                this.player2.clickCard(this.hirumaSignaller);
+                expect(this.player2).toBeAbleToSelect(this.hidaYakamo);
+                this.player2.clickCard(this.hidaYakamo);
+                expect(this.player2).toBeAbleToSelect(this.reprieve);
+                this.player2.clickCard(this.reprieve);
+                expect(this.hidaYakamo.bowed).toBeTruthy();
+                expect(this.hidaYakamo.inConflict).toBeFalsy();
+                expect(this.hirumaSignaller.location).toBe('play area');
+            });
+
+            it('should be able to target itself', function() {
+                this.initiateConflict({
+                    attackers: [this.mirumotoRaitsugu],
+                    defenders: [this.hirumaSignaller, this.hidaYakamo]
+                });
+                this.hirumaSignaller.bow();
+                this.player2.clickCard(this.hirumaSignaller);
+                expect(this.player2).toBeAbleToSelect(this.hirumaSignaller);
+                this.player2.clickCard(this.hirumaSignaller);
+                expect(this.hirumaSignaller.location).toBe('dynasty discard pile');
+                expect(this.spy).toHaveBeenCalledWith('{0} attempted to use {1}, but there are insufficient legal targets', this.player2.player, this.hirumaSignaller);
```

Comment:
For future reference, you can test chat messages with (for example):
```js
expect(this.getChatLogs(1)).toContain('player1 attempted to use Hiruma Signaller, but there are insufficient legal targets')
```
but the way you've done it here is actually a stronger test so please don't change it - I just wanted to let you know a potentially easier way to do it in the future.

Reply:
Thanks for letting me know :)